up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,615 @@ SPDX-FileType: DOCUMENTATION
+----
+sequenceDiagram
+    box White Device1
+        actor uApp
+        participant local uSubscription
+        participant local NotificationTopic
+    end
+    box White Device2
+        participant remote uSubscription
+        participant remote NotificationTopic
+    end
+    
+    uApp->>+local uSubscription: Unsubscribe(UnsubscribeRequest)
+    alt success
+        local uSubscription-->>uApp: Ok
+        local uSubscription-->>local NotificationTopic: Update(UNSUBSCRIBED)
+    else failure
+        local uSubscription-->>uApp: Failure
+    end
+
+    opt last subscription
+        local uSubscription-->>remote uSubscription: Unsubscribe(UnsubscribeRequest)
+        alt success
+            remote uSubscription-->>local uSubscription: Ok
+            remote uSubscription-->>remote NotificationTopic: Update(UNSUBSCRIBED)
+        else failure
+            remote uSubscription-->>local uSubscription: Failure
+        end
+    end
+----
 
-* uSubscription *MUST* change the subscriber to itself (core.usubscription) when unsubscribing to remote topics 
+To allow the reverse flow (publication) to be properly multicast to local subscribers by the local disaptcher when it queries the local uSubscription for a list of local subscribers:
 
+[.specitem,oft-sid="dsn~usubscription-remote_unsubscribe-subscriber-change~1",oft-needs="impl,utest"]
+--
+uSubscription service *MUST* change the subscriber to itself (`core.usubscription`) when unsubscribing to remote topics. 
+--
 
-== Timeout & Retry Logic
+== Open Questions
 
-Subscribe (and unsubscribe) to remote topics are handled by RPC calls between uSubscription services running on the different devices. Given that devices are not always connected to each other, the onus is on uSubscription service to ensure that a command is received in time. Below are the common retry and timeout policies for USubscription service implementations to follow:   
+The following questions are open in the current version of this specification and should be resolved going forward:
 
-* Remote requests *MUST* have a maximum timeout of 5 minutes
-* All timed-out remote commands *MUST* be retied indefinitely until the business logic behind it no longer requires the command to be sent. (ex. the subscriber cals `Unsubscribe()` ) 
-* Remote commands *MUST* be retried upon device to device connectivity (link up) and *MUST NOT* be tried when there is no device connectivity (link down)
+* If a (the last local) client Unsubscribes from a remote topic - should his client-topic relation be set to UNSUBSCRIBED, regardless of whether the remote Unsubscribe call works correctly? What sense would there be in a client-state relation being/remaining UNSUBSCRIBED_PENDING?
+* Implementation-side, seperating client-topic state (what is the subscription state of a specific client wrt a specific topic?) from remote-topic state (what is the global state of a remotely subscribed topic?) makes a lot of sense; put that in the spec?
```

Comment:
Yes agree but it can be implementation specific (not something that requires writing in the spec IMHO))

Reply:
agree